smolyay/surrogate.py

```diff
@@ -1,112 +1,67 @@
+import abc
+import itertools
+import warnings
+
 import numpy
+import sklearn
+import sklearn.base
+import sklearn.linear_model
 
-from smolyay.grid import IndexGridGenerator
+import smolyay
 
 
-class Surrogate:
+class BaseSurrogate(
+    sklearn.base.BaseEstimator,
+    sklearn.base.MultiOutputMixin,
+    sklearn.base.RegressorMixin,
+):
```

Comment:
When we discussed this, we said that the surrogate would not derive from sklearn for the time being. There are some complexities from doing so, I don't think there are many benefits to this for so, and we could always guarantee this later.

Base classes also should not usually have "base" in their name. Please change like this:
```suggestion
class Surrogate:
```